applications/asset_tracker/src/cloud_codec/cloud_codec.c

```diff
@@ -582,6 +582,81 @@ static int cloud_search_cmd(cJSON *root_obj)
 	return 0;
 }
 
+static int cloud_search_config(cJSON * const root_obj)
+{
+	int ret;
+	cJSON *state_obj = NULL;
+	cJSON *config_obj = NULL;
+	cJSON *channel_obj = NULL;
+	struct cmd *chan = NULL;
+	struct cmd const *const group = &group_cfg_set;
+	struct cloud_command found_config_item = {
+		.group = CLOUD_CMD_GROUP_CFG_SET
+	};
```

Comment:
A few of these variables can have tighter scope

Reply:
i thought all variables needed to be declared at the top of the function?

Reply:
I think the "rule" is to put them at the top of the smallest scope. So on top of the inner for-loop if that's the only scope needed for that variable. This may be another rule I've made up, though.
@rlubos Any thoughts?

Reply:
I saw this rule being enforced in upstream Zephyr, I'm not aware if it's written down. Still, I'm convinced that narrowing scope is a very good practice.